subsys/mpsl/fem/common/mpsl_fem_utils.c

```diff
@@ -5,14 +5,41 @@
  */
 
 #include <mpsl_fem_utils.h>
-
-#if defined(CONFIG_MPSL_FEM_PIN_FORWARDER)
-
 #include <stdbool.h>
 #include <string.h>
 #include <zephyr/devicetree.h>
 #include <zephyr/sys/__assert.h>
 
+#if !defined(CONFIG_MPSL_FEM_PIN_FORWARDER)
+
+void mpsl_fem_extended_pin_to_mpsl_fem_pin(uint8_t pin_num, mpsl_fem_pin_t* p_fem_pin)
+{
+    if (pin_num < 32) {
+#if !DT_NODE_HAS_STATUS(DT_NODELABEL(gpio0), okay)
+        __ASSERT(false, "Unknown GPIO port DT label");
+#else
+        p_fem_pin->port_pin = pin_num;
+        p_fem_pin->p_port   =  (NRF_GPIO_Type *) DT_REG_ADDR(DT_NODELABEL(gpio0));
+        p_fem_pin->port_no  = 0;
+#endif
+        return;
+    }
+    if (pin_num < 64) {
+#if !DT_NODE_HAS_STATUS(DT_NODELABEL(gpio1), okay)
+        __ASSERT(false, "Unknown GPIO port DT label");
+#else
+        p_fem_pin->port_pin = pin_num - 32;
+        p_fem_pin->p_port   =  (NRF_GPIO_Type *) DT_REG_ADDR(DT_NODELABEL(gpio1));
+        p_fem_pin->port_no  = 1;
+#endif
+        return;
```

Comment:
Well. there is `nrf_gpio_pin_port_decode` in nrf_gpio.h I've changed it, you can see it. However I'm not 100% convinced if it is better. It will be slower and not really more readable, as there is no function which would get the port and not overwrite the current pin. Also it will probably use more space in flash. I have saved the old version, take a look and see if the new is better.